libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1482,7 +1482,16 @@ void TypeChecker::typeCheckABIEncodeFunctions(
 
 		if (argType->category() == Type::Category::RationalNumber)
 		{
-			if (!argType->mobileType())
+			auto const& rationalType = dynamic_cast<RationalNumberType const&>(*argType);
+			if (rationalType.isFractional())
+			{
+				m_errorReporter.typeError(
+					arguments[i]->location(),
+					"Fixed point numbers cannot yet be encoded."
```

Comment:
Perhaps just mention "fractional"? The fact that the number is implicitly converted to a fixed point number might not be obivous to the user.

Reply:
Done